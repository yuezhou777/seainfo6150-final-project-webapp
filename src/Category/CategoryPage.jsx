import React from 'react';
import NavBar from "../NavBar/NavBar";
import styles from './CategoryPage.module.css';
import CategoryList from "../CategoryList/CategoryList";

const CategoryPage = (props) => {

    const allCategory = new Set();

    if (props.posts.length) {
        props.posts.map(post => (
            allCategory.add(post.breed)
        ))
    }

    let categoryList = [];
    allCategory.forEach((key, value) => (
        categoryList.push(value)
    ))

    return(

        <div>

            <div>
                <NavBar />
            </div>

            <div className={styles.categoryBoard}>
                <CategoryList breeds={categoryList} posts={props.posts} />
            </div> 

        </div>

    );

}

export default CategoryPage;