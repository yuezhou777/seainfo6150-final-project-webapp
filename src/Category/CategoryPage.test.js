import React from "react";
import { render } from "@testing-library/react";
import CategoryPage from "./CategoryPage.jsx";
import { BrowserRouter } from 'react-router-dom';

it("renders HomePage Correctly", () => {

    const posts = [
        {
            "id":"1",
            "title":"Mel",
            "intro":"The only thing better than one adorable husky is an entire sled team of 'em. ",
            "content":"Mel is a gentle and sweet dog. She takes food out of your hand gently and likes to give little kisses, she still likes to run and play as if she was a puppy. Once she gets to know someone, she becomes less shy and more affectionate. She enjoys staying by a person's side.",
            "age":"1 year",
            "sex":"Female",
            "image":{
               "_url":"https://images.unsplash.com/photo-1605559141066-1549783d18e9?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=400&q=250",
               "size":123
            },
            "breed":"Husky",
            "category":"dog"
        }
    ];

    const { container } = render(
        <BrowserRouter>
            <CategoryPage posts={posts}/>
        </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
});