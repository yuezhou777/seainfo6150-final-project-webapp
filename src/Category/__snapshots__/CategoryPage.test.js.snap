// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HomePage Correctly 1`] = `
<div>
  <div>
    <div>
      <div
        class="navContainer"
      >
        <header
          class="navContainer"
        >
          <nav>
            <a
              class="navIcon"
              href="/"
            >
              PurrPet
            </a>
          </nav>
          <nav>
            <ul
              class="navList"
            >
              <li
                class="navItem"
              >
                <a
                  class="link"
                  href="/aboutus"
                >
                  About Us
                </a>
              </li>
              <li
                class="navItem"
              >
                <a
                  class="link"
                  href="/allpets"
                >
                  All Pets
                </a>
              </li>
              <li
                class="navItem"
              >
                <a
                  class="link"
                  href="/category"
                >
                  Categories
                </a>
              </li>
              <li
                class="navItem"
              >
                <a
                  class="link"
                  href="/contactus"
                >
                  Contact Us
                </a>
              </li>
            </ul>
          </nav>
        </header>
      </div>
    </div>
    <div
      class="categoryBoard"
    >
      <div
        class="container"
      >
        <h3
          class="categoryTitle"
        >
          All Pet Categories
        </h3>
        <ul
          class="listContainer"
        >
          <h4>
            <a
              class="category"
              href="/category/Husky"
            >
              Husky
            </a>
          </h4>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
